### PR TITLE
Order methods test

### PR DESCRIPTION
**Pull request by michael-alade**
**Resolves**
#2095 

**Test**
- Run `reaction test`

### CODE REVIEW

File: lib/collections/schemas/orders.js [hunk 1]
```diff
@@ -23,6 +23,9 @@ export const History = new SimpleSchema({
   event: {
     type: String
   },
+  value: {
```
Comment (zenweasel) [inline lib/collections/schemas/orders.js]: Why?

File: imports/plugins/core/dashboard/client/templates/shop/settings/settings.html [hunk 2]
```diff
@@ -12,6 +12,7 @@
 <template name="shopSettings">
 
   <div class="panel-group" id="shopSettingsAccordian" role="tablist" aria-multiselectable="true">
+    {{> shopBrandImageOption}}
```
Comment (zenweasel) [inline imports/plugins/core/dashboard/client/templates/shop/settings/settings.html]: Why?

File: server/methods/core/orders.app-test.js [hunk 3]
```diff
@@ -61,16 +74,26 @@ describe("orders test", function () {
...
+      });
+    });
+
     it("should return an error if user is not admin", function () {
       sandbox.stub(Reaction, "hasPermission", () => false);
       const returnToStock =  false;
       spyOnMethod("cancelOrder", order.userId);
 
-      function copyStartCancelOrder() {
+      function CancelOrder() {
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: function names already start with a lowercase letter.

File: server/methods/core/orders.app-test.js [hunk 4]
```diff
@@ -121,4 +144,358 @@ describe("orders test", function () {
...
+      const orderBilling = Orders.findOne({ _id: order._id }).billing[0];
+      expect(orderBilling.paymentMethod.status).to.equal("approved");
+      expect(orderBilling.paymentMethod.mode).to.equal("capture");
+      expect(orderBilling.invoice.discounts).to.equal(discount);
+      expect(orderBilling.invoice.total).to.equal(Number(total));
+    });
+  });
+
+  describe("orders/shipmentShipped", function () {
+    it("should throw an error does not have permission", function () {
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: this description needs to be at least grammatically complete

File: server/methods/core/orders.app-test.js [hunk 5]
```diff
@@ -121,4 +144,358 @@ describe("orders test", function () {
...
+      sandbox.stub(Media, "findOne", () => {
+        // stub url method for media file
+        const url = () => "/stub/url";
+        return {
+          url
+        };
+      });
+      sandbox.stub(Shops, "findOne", () => shop);
+      const result = Meteor.call("orders/sendNotification", order);
+      expect(result).to.equal(true);
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: You should probably use the more readable `expect(result).to.be.true`

File: server/methods/core/orders.app-test.js [hunk 6]
```diff
@@ -121,4 +144,358 @@ describe("orders test", function () {
...
+      sandbox.stub(Reaction, "hasPermission", () => false);
+      spyOnMethod("updateShipmentTracking", order.userId);
+      function updateShipmentTracking() {
+        const trackingValue = "2340FLKD104309";
+        return Meteor.call("orders/updateShipmentTracking", order, order.shipping[0], trackingValue);
+      }
+      expect(updateShipmentTracking).to.throw(Meteor.error, /Access Denied/);
+    });
+
+    it("should the tracking value", function () {
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: The description needs help

File: server/methods/core/orders.app-test.js [hunk 7]
```diff
@@ -121,4 +144,358 @@ describe("orders test", function () {
...
+    it("should return access denied if user does not have access", function () {
+      sandbox.stub(Reaction, "hasPermission", () => false);
+      spyOnMethod("capturePayments", order.userId);
+      function capturePayments() {
+        return Meteor.call("orders/capturePayments", order._id);
+      }
+      expect(capturePayments).to.throw(Meteor.error, /Access Denied/);
+    });
+
+    it("should update the ite workflow to coreOrderItemWorkflow/captured", function () {
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: Typo in description

File: server/methods/core/orders.app-test.js [hunk 8]
```diff
@@ -121,4 +144,358 @@ describe("orders test", function () {
...
+      return done();
+    });
+  });
+
+  describe("orders/refunds/list", function () {
+    it("should return an array of refunds", function () {
+      sandbox.stub(Reaction, "hasPermission",  () => true);
+      spyOnMethod("refunds/list", order.userId);
+      Meteor.call("orders/refunds/list", order, (err, res) => {
+        expect(typeof res).to.equal(Array);
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: So all this does is check that the method returns an empty array?

File: server/methods/core/orders.js [hunk 9]
```diff
@@ -443,7 +402,7 @@ export const methods = {
     this.unblock();
 
     // Get Shop information
-    const shop = Shops.findOne(order.shopId);
+    const shop = Shops.findOne({ _id: order.shopId });
```
Comment (zenweasel) [inline server/methods/core/orders.js]: The original form is actually more correct

File: lib/collections/schemas/orders.js (hunk 1 above)
Comment (michael-alade) [inline lib/collections/schemas/orders.js]: A tracking value is supposed to be added to the value key.

File: server/methods/core/orders.app-test.js [hunk 10]
```diff
@@ -121,4 +144,359 @@ describe("orders test", function () {
...
+      sandbox.stub(Reaction, "hasPermission", () => false);
+      spyOnMethod("updateShipmentTracking", order.userId);
+      function updateShipmentTracking() {
+        const trackingValue = "2340FLKD104309";
+        return Meteor.call("orders/updateShipmentTracking", order, order.shipping[0], trackingValue);
+      }
+      expect(updateShipmentTracking).to.throw(Meteor.error, /Access Denied/);
+    });
+
+    it("should the update the order tracking value", function () {
```
Comment (zenweasel) [inline server/methods/core/orders.app-test.js]: Just `should update` not `should the`

**Pull request merged**